---
title: "p2_PCAngsd_angsd1"
author: "Julia Harenčár"
date: "5/27/2021"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.width = 7, fig.height = 5, fig.path = 'PCAngsd_figs/')
knitr::opts_knit$set(root.dir = '/Users/juliaharencar/Google Drive (jharenca@ucsc.edu)/GitHub/seq_analysis/ANGSD')
```
 This markdown summarizes the PCAs I generated by running PCAngsd on the plate 2 genotype likelihood (beagle) file output of angsd run 1, which uses samtools as the variant caller (see AVHonly_ANGSD1_beagles.slurm in angsd_runs/).
 
#### PCAs of alle, vill, and hybrids only (plate 2)

PCAngsd run with estimation of independent allele frequencies.
Plot created from output of the following code:\
`pcangsd.py -beagle $IN_DIR/p2_AVH.genolike.beagle.gz -o $OUT_DIR/AVH_p2_PCAngsd_IAF`


```{r angsd1_AVH_PCA_IAF, echo=FALSE}

pop <- read.table("angsd_out/AVH_only.pop.info")
subpop <- read.table("angsd_out/AVH_only.subpop.info")
cov <- as.matrix(read.table("angsd_out/PCAngsd/AVH_only_PCAngsd_IAF.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- pop[,1]
spp.site <- subpop[,1]
PC12.AVH.IAF <- data.frame(PC1, PC2, spp, spp.site)

library(ggplot2)
ggplot(PC12.AVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp))) +
  ggtitle("AVH.IAF")

ggplot(PC12.AVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp.site))) +
  ggtitle("AVH.IAF.w/site")
  
```

