---
title: "PCAngsd_angsd1"
author: "Julia Harenčár"
date: "2/25/2021"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.width = 7, fig.height = 5, fig.path = 'PCAngsd_figs/')
knitr::opts_knit$set(root.dir = '/Users/juliaharencar/Google Drive (jharenca@ucsc.edu)/GitHub/seq_analysis/ANGSD')
```
 This markdown summarizes the PCAs I generated by running PCAngsd on the genotype likelihood (beagle) file output of angsd run 1, which uses samtools as the variant caller (see AVHonly_ANGSD1_beagles.slurm in angsd_runs/).
 
#### PCAs of alle, vill, and hybrids only

PCAngsd run with estimation of independent allele frequencies.
Plot created from output of the following code:\
`pcangsd.py -beagle $IN_DIR/AVH_only.genolike.beagle.gz -o $OUT_DIR/AVH_only_PCAngsd_IAF`


```{r angsd1_AVH_PCA_IAF, echo=FALSE}

pop <- read.table("angsd_out/AVH_only.pop.info")
subpop <- read.table("angsd_out/AVH_only.subpop.info")
cov <- as.matrix(read.table("angsd_out/PCAngsd/AVH_only_PCAngsd_IAF.cov"))
e <- eigen(cov)

PC1 <- e$vectors[,1]
PC2 <- e$vectors[,2]
spp <- pop[,1]
spp.site <- subpop[,1]
PC12.AVH.IAF <- data.frame(PC1, PC2, spp, spp.site)

library(ggplot2)
ggplot(PC12.AVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp))) +
  ggtitle("AVH.IAF")

ggplot(PC12.AVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp.site))) +
  ggtitle("AVH.IAF.w/site")
  
```


PCAngsd run without estimation of independent allele frequencies.
Plot created from output of the following code:\
`pcangsd.py -beagle $IN_DIR/AVH_only.genolike.beagle.gz -o $OUT_DIR/AVH_only_PCAngsd -iter 0`


```{r angsd1_AVH_PCA, echo=FALSE}

pop2<-read.table("angsd_out/AVH_only.pop.info")
subpop2 <- read.table("angsd_out/AVH_only.subpop.info")
cov2 <- as.matrix(read.table("angsd_out/PCAngsd/AVH_only_PCAngsd.cov"))
e2<-eigen(cov2)

PC1 <- e2$vectors[,1]
PC2 <- e2$vectors[,2]
spp <- pop2[,1]
spp.site <- subpop2[,1]
PC12.AVH <- data.frame(PC1, PC2, spp, spp.site)

library(ggplot2)
ggplot(PC12.AVH, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp))) +
  ggtitle("AVH.noIAF")

ggplot(PC12.AVH, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp.site))) +
  ggtitle("AVH.noIAF.w/site")
  
```

#### PCAs of laev, guan, alle, vill, and alle x vill hybrids.

PCAngsd run with estimation of independent allele frequencies.
Plot created from output of the following code:\
`pcangsd.py -beagle $IN_DIR/L.G.A.V.H.genolike.beagle.gz -o $OUT_DIR/L.G.A.V.H_PCAngsd_IAF`

```{r angsd1_LGAVH_PCA_IAF, echo=FALSE}

pop3<-read.table("angsd_out/LGAVH.pop.info")
subpop3 <- read.table("angsd_out/LGAVH.subpop.info")
cov3 <- as.matrix(read.table("angsd_out/PCAngsd/L.G.A.V.H_PCAngsd_IAF.cov"))
e3<-eigen(cov3)

PC1 <- e3$vectors[,1]
PC2 <- e3$vectors[,2]
spp <- pop3[,1]
spp.site <- subpop3[,1]
PC12.LGAVH.IAF <- data.frame(PC1, PC2, spp, spp.site)

library(ggplot2)
ggplot(PC12.LGAVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp))) +
  ggtitle("LGAVH.IAF")

ggplot(PC12.LGAVH.IAF, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp.site))) +
  ggtitle("LGAVH.IAF.w/site")
  
```


PCAngsd run without estimation of independent allele frequencies.
Plot created from output of the following code:\
`pcangsd.py -beagle $IN_DIR/L.G.A.V.H.genolike.beagle.gz -o $OUT_DIR/L.G.A.V.H_PCAngsd -iter 0`

```{r angsd1_LGAVH_PCA, echo=FALSE}

pop4<-read.table("angsd_out/LGAVH.pop.info")
subpop4 <- read.table("angsd_out/LGAVH.subpop.info")
cov4 <- as.matrix(read.table("angsd_out/PCAngsd/L.G.A.V.H_PCAngsd.cov"))
e4<-eigen(cov4)

PC1 <- e4$vectors[,1]
PC2 <- e4$vectors[,2]
spp <- pop4[,1]
spp.site <- subpop4[,1]
PC12.LGAVH <- data.frame(PC1, PC2, spp, spp.site)

library(ggplot2)
ggplot(PC12.LGAVH, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp)))  +
  ggtitle("LGAVH.noIAF")

ggplot(PC12.LGAVH, aes(x=PC1, y=PC2)) +
  geom_point(aes(color=factor(spp.site)))  +
  ggtitle("LGAVH.noIAF.w/site")
  
```
